pkg/clusterapi/system.go

```diff
@@ -212,6 +214,15 @@ func (c *system) Run(ctx context.Context) error { //nolint:gocyclo
 			return fmt.Errorf("unable to retrieve azure session: %w", err)
 		}
 
+		b, err := json.Marshal(session.Environment)
+		if err != nil {
+			return errors.Wrap(err, "could not serialize Azure Stack endpoints")
+		}
+		ashFN := filepath.Join(c.componentDir, "azurestackcloud.json")
+		if err = os.WriteFile(ashFN, b, 0600); err != nil {
+			return fmt.Errorf("failed to write Azure Stack environment file: %w", err)
+		}
```

Comment:
Should this block be conditional - for AzureStack only? We probably can set `AZURE_ENVIRONMENT_FILEPATH` to `""` on non-azurestack environment.

Reply:
done